File: ruby/testcode/authzfailure_013.rb
require_relative 'shared'

# vuln-code-snippet start ruby_authz_unscoped_update
def update_record(req)
  record_id = req.param('id')
  value = req.post('value')
  BenchmarkResponse.ok("record #{record_id} updated to #{value}") # vuln-code-snippet vuln-line ruby_authz_unscoped_update
end
# vuln-code-snippet end ruby_authz_unscoped_update

File: rust/testcode/authzfailure_013.rs
//! CWE-285: User-controlled admin cookie trusted for delete authorization

fn delete_record(id: &str) -> String {
    format!("record_{}_deleted", id)
}

// vuln-code-snippet start testcodeAuthzfailure013
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let id = req.param("id");
    let is_admin = req.cookie("admin") == "true";
    if is_admin {
        let result = delete_record(&id); // vuln-code-snippet target-line testcodeAuthzfailure013
        return super::shared::BenchmarkResponse::ok(&result);
    }
    super::shared::BenchmarkResponse::forbidden("access denied")
}
// vuln-code-snippet end testcodeAuthzfailure013
